django/thunderstore/frontend/api/experimental/views/frontpage.py

```diff
@@ -0,0 +1,104 @@
+from typing import Optional
+
+from django.db.models import Sum
+from drf_yasg.utils import swagger_auto_schema
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from thunderstore.community.models.community import Community
+from thunderstore.frontend.api.experimental.serializers.views import (
+    FrontPageContentSerializer,
+)
+from thunderstore.repository.models import Package, PackageVersion
+
+
+class FrontPageApiView(APIView):
+    """
+    Return information required to render the site's front page.
+    """
+
+    serializer_class = FrontPageContentSerializer
+    permission_classes = []
+
+    @swagger_auto_schema(
+        responses={200: FrontPageContentSerializer()},
+        operation_id="experimental.frontend.frontpage",
+    )
+    def get(self, request, *args, **kwargs):
+        community_objs = Community.objects.listed().prefetch_related(
+            "package_listings__package__versions",
+            "sites",
+        )
+        communities = [
+            {
+                "downloadCount": get_community_downloads(c),
+                "imageSrc": get_site_image(c),
+                "modCount": get_community_mods(c),
+                "name": c.name,
+            }
+            for c in community_objs
+        ]
+        communities.sort(key=lambda c: c["modCount"], reverse=True)
+
+        serializer = FrontPageContentSerializer(
+            {
+                "communities": communities,
+                "downloadCount": get_downloads_grand_total(),
+                "modCount": get_mods_grand_total(),
+            },
+        )
+
+        return Response(serializer.data, status=status.HTTP_200_OK)
+
+
+def get_community_downloads(community: Community) -> int:
+    """
+    Return sum of downloads of packages listed under the community.
+
+    A package can be listed under multiple communities. Downloading it
+    from any community will increase the download count in all
+    communities where it's listed.
+    """
+    versions = [
+        v for pl in community.package_listings.all() for v in pl.package.versions.all()
+    ]
+    return sum(version.downloads for version in versions)
+
+
+def get_community_mods(community: Community) -> int:
+    """
+    Return number of mods published under the given community.
+    """
+    return community.package_listings.count()
+
+
+def get_downloads_grand_total() -> int:
+    """
+    Return sum of all downloads of all package versions.
+
+    Counts mods included in multiple communities only once.
+    """
+    result = PackageVersion.objects.aggregate(Sum("downloads"))
+    return result["downloads__sum"] or 0
+
+
+def get_mods_grand_total() -> int:
```

Comment:
I would combine this & `get_downloads_grand_total` to a single query:

```python
result = Package.objects.annotate(count=Value(1)).aggregate(
    downloads_total=Sum("downloads"),
    count=Sum("count", output_field=models.BigIntegerField()),
)
```

You might even be able to skip the annotate and just do


```python
result = Package.objects.aggregate(
    downloads_total=Sum("downloads"),
    count=Value(1, output_field=models.BigIntegerField()),
)
```